lib/rules/no-extra-parens.js

```diff
@@ -168,6 +168,18 @@ module.exports = {
         }
 
         /**
+         * Determines if a constructor function is newed-up with parens
+         * @param {ASTNode} newExpression - The NewExpression node to be checked.
+         * @returns {boolean} True if the constructor is called with parens.
+         * @private
+         */
+        function isNewExpressionWithParens(newExpression) {
+            const lastToken = sourceCode.getLastToken(newExpression);
+
+            return lastToken.value === ")";
```

Comment:
This function will incorrectly return `true` for

```js
new (Foo)
```

As a result, the rule still incorrectly reports the following code, even with this patch:

```js
(new (Foo || Bar))()
```

I think the correct way to do this is to check the second-to-last token as well, like the implementation in [new-parens](https://github.com/eslint/eslint/blob/df2351a695462d9490f665119a7153157bd16ef3/lib/rules/new-parens.js).